src/components/ChatBeginningText.js

```diff
@@ -0,0 +1,89 @@
+import React, {Fragment} from 'react';
+import PropTypes from 'prop-types';
+import lodashGet from 'lodash/get';
+import Str from 'expensify-common/lib/str';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import styles from '../styles/styles';
+import Text from './Text';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import compose from '../libs/compose';
+import {getPersonalDetailsForLogins} from '../libs/OptionsListUtils';
+import ONYXKEYS from '../ONYXKEYS';
+
+const propTypes = {
+    /** Wether it is a default Chat Room */
+    isDefaultChatRoom: PropTypes.bool,
+
+    /** The report currently being looked at */
+    report: PropTypes.oneOfType([PropTypes.object]),
+
+    ...withLocalizePropTypes,
+
+};
+
+const defaultProps = {
+    report: {},
+    isDefaultChatRoom: false,
+};
+
+const ChatBeginningText = ({
+    isDefaultChatRoom, report, translate, personalDetails, toLocalPhone,
+}) => {
+    const participants = lodashGet(report, 'participants', []);
+    const isMultipleParticipant = participants.length > 1;
+    const displayNamesWithTooltips = _.map(
+        getPersonalDetailsForLogins(participants, personalDetails),
+        ({
+            displayName, firstName, login, pronouns,
+        }) => {
+            const displayNameTrimmed = Str.isSMSLogin(login) ? toLocalPhone(displayName) : displayName;
+
+            return {
+                displayName: (isMultipleParticipant ? firstName : displayNameTrimmed) || Str.removeSMSDomain(login),
+                tooltip: Str.removeSMSDomain(login),
+                pronouns,
+            };
+        },
+    );
+    const chatUsers = isDefaultChatRoom ? [{displayName: report.reportName}] : displayNamesWithTooltips;
+
+    return (
+        <Text style={[styles.mt3, styles.w70, styles.textAlignCenter]}>
+            <Text>
+                {isDefaultChatRoom
+                    ? `${translate('reportActionsView.beginningOfChatHistoryPrivate', {name: lodashGet(chatUsers, '[0].displayName', '')})}`
+                    : `${translate('reportActionsView.beginningOfChatHistory')} `}
+            </Text>
+            {!isDefaultChatRoom
+            && (
+                <>
+                    {chatUsers.map(({displayName, pronouns}, index) => (
+                        <Text key={displayName}>
+                            <Text style={[styles.textStrong]}>
+                                {displayName}
+                            </Text>
+                            {Boolean(pronouns) && <Text>{` (${pronouns})`}</Text>}
+                            {(chatUsers.length === 1 || chatUsers.length - 1 === index) && '.'}
```

Comment:
A few suggestions here:

1. I think using a boolean expression like this to render a string can crash mobile unless you wrap the string in a text component.
1. There's a few redundant conditionals here

```javascript
{(index === chatUsers.length - 1) && <Text>.</Text>}
{(index === chatUsers.length - 2) && <Text>` ${translate('common.and')} `</Text>}
{(index < chatUsers.length - 2) && <Text>', '</Text>}
```

Reply:
Also, I think it would be very useful to just create a single `TextList` component that takes in a set of `listItems`, each providing an optional style, and that chains them together into a string list component like this.

Reply:
thanks a lot, I updated the code and I don't think so there is any need for a component if you want the component please suggest the code.